plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java

```diff
@@ -1414,6 +1414,10 @@ private void configureAgentHooks(final Map<String, Object> params) {
         s_logger.debug("agent.hooks.libvirt_vm_on_stop.method is " + _agentHooksVmOnStopMethod);
     }
 
+    public boolean isUefiPropertiesFileSet() {
```

Comment:
@nvazquez UEFI properties are loaded from this file while configuring the resource, no need to check again for this file, instead check for properties loaded or not. Also, re-use this method in _loadUefiProperties()_ method below for the same condition. Thanks.

```suggestion
    public boolean isUefiPropertiesLoaded() {
        return _uefiProperties != null && _uefiProperties.getProperty("guest.loader.legacy") != null
    }
```

Reply:
Done, thanks - nice catch